src/inventory_ui.cpp

```diff
@@ -401,14 +454,329 @@ void inventory_selector::print_right_column() const
+            }
+            // End Cell Border Stuff
+        }
+        
+        std::string item_name = it.label(); // Get object's display name
+        size_t count = 1;
+        if(cur_entry.slice != NULL) {
+            count = cur_entry.slice->size();
+            if(count > 1) {
+                item_name = it.label(count);
+            }
+        }
+        
+        if(curr_line >= 2 && curr_line + 1 < TERMY) {  // Second line
+            // Cell Border Stuff
+            if(row_i == 0) {
+                mvwputch(w_inv, curr_line + 1, 1, c_white, LINE_XOXO);
+            }
+            for(char x = 2; x < 26; x++) {
+                mvwputch(w_inv, curr_line + 1, x + (row_i * 25), fill, ' ');
+            }
+            mvwputch(w_inv, curr_line + 1, 26 + (row_i * 25), c_white, LINE_XOXO);
+            // End Cell Border Stuff
+        
+            if(it.invlet != 0) { // Show item hotkey (if any)
+                mvwputch(w_inv, curr_line + 1, 2 + (row_i * 25), invlet_color, it.invlet);
+            }
+            
+            if(count > 1) { // Show item count (if more than one)
+                mvwprintz(w_inv, curr_line + 1, 3 + (row_i * 25), fill, "%22ux", count);
+            }
+            
+            if(OPTIONS["ITEM_HEALTH_BAR"] && (it.damage != 0 || it.is_armor()) && !it.is_null()) { // Show item damage (if appropriate)
+                auto const &nc_text = get_item_hp_bar(it.damage);
+                mvwprintz(w_inv, curr_line + 1, 6 + (row_i * 25), nc_text.second, nc_text.first.c_str());
+            }
+            
+            mvwprintz(w_inv, curr_line + 1, 10 + (row_i * 25), fill, it.get_mod_string().c_str()); // Show mod count (if any)
+        }
+        
+        std::string temp[2];
+        std::string *name_lines = split_name(item_name, temp);
+        
+        if(curr_line >= 1 && curr_line + 2 < TERMY) {  // Third line
+            // Cell Border Stuff
+            if(row_i == 0) {
+                mvwputch(w_inv, curr_line + 2, 1, c_white, LINE_XOXO);
+            }
+            for(char x = 2; x < 26; x++) {
+                mvwputch(w_inv, curr_line + 2, x + (row_i * 25), fill, ' ');
+            }
+            mvwputch(w_inv, curr_line + 2, 26 + (row_i * 25), c_white, LINE_XOXO);
+            // End Cell Border Stuff
+            
+            mvwprintz(w_inv, curr_line + 2, 2 + (row_i * 25), name_color, name_lines[0].c_str()); // Print line one of name
+        }
+        
+        if(curr_line >= 0 && curr_line + 3 < TERMY) {  // Fourth line
+            // Cell Border Stuff
+            if(row_i == 0) {
+                mvwputch(w_inv, curr_line + 3, 1, c_white, LINE_XOXO);
+            }
+            for(char x = 2; x < 26; x++) {
+                mvwputch(w_inv, curr_line + 3, x + (row_i * 25), fill, ' ');
+            }
+            mvwputch(w_inv, curr_line + 3, 26 + (row_i * 25), c_white, LINE_XOXO);
+            // End Cell Border Stuff
```

Comment:
I haven't counted it, but this block ("Cell Border Stuff") seems the be repeated several times. It should go into a function / local lambda.